src/socket/src/sink.rs

```diff
@@ -9,7 +9,7 @@ use async_mutex::Mutex;
 use async_mutex::MutexGuard;
```

Comment:
minor version update for for this crate

Reply:
I think we shouldn't do that in this PR to avoid a merge conflict of mine. I can't find a specific announcement bot the [`async-mutex`](https://crates.io/crates/async-mutex) github points to [`async-lock`](https://crates.io/crates/async-lock). I know because the wasm client needs the latest a more recent version of `async-lock` and results in https://github.com/rust-lang/rust/issues/48564 which was fixed in various parts and added to

Reply:
@simlay  How about if I update to use async-mutex to async-lock as first step?

Reply:
Sure. I'll do that.

Reply:
Done in https://github.com/infinyon/fluvio/pull/1086